examples/mlx_metal_kernel_opt/evaluator.py

```
@@ -0,0 +1,1395 @@
+                    try:
+                        # Clean GPU state
+                        self._ensure_clean_gpu_state()
+
+                        # Create conservative test inputs
+                        x = mx.random.normal((B, L, D)) * 0.1  # Smaller values for safety
+                        mask = "causal"
+
+                        # Test with maximum GPU protection
+                        success, result = self._bulletproof_execute_with_gpu_protection(
+                            lambda: self._test_single_sequence_memory_safe(
+                                custom_attention_class, args, x, mask
```

Ruff (0.11.9)

Function definition does not bind loop variable `x`

(B023)